### ISSUE TITLE
`Type[Any]` does not unify with `type`

### ISSUE THREAD

**Issue by wbadart** [microsoft/pyright#1806]
**Describe the bug**

When a function expects an argument of type `type` but is supplied a value whose type is inferred to be `Type[Any]`, a type error is reported.

**To Reproduce**

```python
import dataclasses

def foo(t1: type) -> type:
    return dataclasses.make_dataclass(
        cls_name="foo",
        fields=[(f.name, f.type, f) for f in dataclasses.fields(t1)],
    )
```

```
[Pyright reportGeneralTypeIssues] [E] Argument of type "list[tuple[str, Type[Any], Field[Any]]]" cannot be assigned to parameter "fields" of type "Iterable[str | Tuple[str, type] | Tuple[str, type, Field[Any]]]" in function def suppl "make_dataclass"
```

**Expected behavior**

The documentation for [`typing.Type`](https://docs.python.org/3/library/typing.html#typing.Type) mentions that `Type`, `Type[Any]`, and `type` are equivalent, so it seems that `list[tuple[str, Type[Any], Field[Any]]]` should be compatible with `Iterable[Tuple[str, type, Field[Any]]]`, and I'd expect there not be a type error in the above snippet.

**VS Code extension or command-line**

Using `v1.1.109` in neovim via [`coc-pyright`](https://github.com/fannheyward/coc-pyright).

Thanks!

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.137, which I just published. It will also be included in the next release of pylance.

**Comment by wbadart** [comment]
Awesome, thanks Eric
